File: 콜라츠추측.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int num) {
    int answer = 0;
    
    long long a = num;
    while(a != 1){
        if(a % 2 == 0) a = a/ 2;
        else a = a * 3 + 1;
        ++answer;
        if(answer >499) return -1;
    }
    
    return answer;
}

File: python/콜라츠추측.py
def solution(num):
    num = float(num)
    count = 0

    #& 과 and 는 다르다!!!!!!!
    #and 는 && 이고 & 는 비트 연산자라고 보면된다!
    while num != 1 and count < 500:
        count += 1
        if not num % 2:
            num /= 2
        else :
            num = num * 3 + 1
            
    return -1 if count >= 500 else count